core/genesis.go

```diff
@@ -275,15 +276,34 @@ func (g *Genesis) toBlock(db ethdb.Database, triedb *triedb.Database) *types.Blo
 	if g.Difficulty == nil {
 		head.Difficulty = params.GenesisDifficulty
 	}
+	if g.ExtraData == nil {
+		head.Extra = []byte{}
+	}
 	if conf := g.Config; conf != nil {
 		num := new(big.Int).SetUint64(g.Number)
-		if params.GetExtra(conf).IsApricotPhase3(g.Timestamp) {
+		confExtra := params.GetExtra(conf)
+		if confExtra.IsApricotPhase3(g.Timestamp) {
 			if g.BaseFee != nil {
 				head.BaseFee = g.BaseFee
 			} else {
 				head.BaseFee = big.NewInt(ap3.InitialBaseFee)
 			}
 		}
+		headerExtra := customtypes.GetHeaderExtra(head)
+
+		// When Etna/Cancun is active, `BlockGasCost` and `ExtDataGasUsed` are decoded to 0 if it's nil.
+		// This is because these fields come before than other optional Cancun fields in RLP order.
```

Comment:
There's a typo in this line. I think this is correct:
```suggestion
		// This is because these fields come before the other optional Cancun fields in RLP order.
```